util/config/compileline

```diff
@@ -1,154 +1,201 @@
+  parser.add_option("--llvm",
+                    dest="llvm", action='store_true', default=False,
+                    help="specify that the rest of the arguments are for a"
+                         "compilation working with the LLVM backend. This "
+                         "argument causes this script to set "
+                         "CHPL_TARGET_COMPILER=clang-included")
+  parser.add_option("--compile", "--compile-cc",
+                    dest="compilecc", action='store_true', default=False,
+                    help="print a C compiler invocation that can use"
+                         "the Chapel runtime")
+  parser.add_option("--compile-c++",
+                    dest="compilecxx", action='store_true', default=False,
+                    help="print a C++ compiler invocation that can use"
+                         "the Chapel runtime")
+  parser.add_option("--cflags",
+                    dest="cflags", action='store_true', default=False,
+                    help="print C compilation flags that Chapel would use")
+  parser.add_option("--c++flags",
+                    dest="cxxflags", action='store_true', default=False,
+                    help="print C++ compilation flags that Chapel would use")
+  parser.add_option("--includes-and-defines",
+                    dest="includesanddefines", action='store_true', default=False,
+                    help="Print -D and -I compilation flags to include"
+                         "the Chapel runtime")
+  parser.add_option("--libraries",
+                    dest="libraries", action='store_true', default=False,
+                    help="print out -L and -l options to link with"
+                         "the Chapel runtime")
+  parser.add_option("--main.o",
+                    dest="maino", action='store_true', default=False,
+                    help="pront out the path to the main.o file")
+  parser.add_option("--llvm-install-dir",
+                    dest="llvminstalldir", action='store_true', default=False,
+                    help="print path to included LLVM installation")
+  parser.add_option("--clang",
+                    dest="clangcc", action='store_true', default=False,
+                    help="print which clang C compiler is configured")
+  parser.add_option("--clang++",
+                    dest="clangcxx", action='store_true', default=False,
+                    help="print which clang C++ compiler is configured")
+  parser.add_option("--clang-sysroot-arguments",
+                    dest="clangsysroot", action='store_true', default=False,
+                    help="print out any saved clang arguments that"
+                         "specify the system root")
+
+  (options, args) = parser.parse_args()
+
+  # Handle VAR=VAL environment variable setting
+  for arg in args:
+    kv = arg.split("=", 2)
+    if len(kv) == 2:
+      os.environ[kv[0]] = os.environ[kv[1]]
+
+  make_helper = make + ["-f", chpl_home_dir + "/runtime/etc/Makefile.include"]
```

Comment:
If you find it cleaner to maintain commands as strings (I usually do), you can wrap the `run_command()` args with `shlex.split` instead:

```python
import shlex

...

def myrun(cmd):
    output = utils.run_command(shlex.split(cmd))
    ...

# e.g.
myrun('make -f ' + chpl_home_dir + '/runtime/etc/Makefile.include']

Reply:
I'm OK with the lists-of-strings for this one